src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,184 @@
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void addUserWhoAlreadyExist_NotOk() {
+        registrationService.register(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin(DEFAULT_LOGIN);
+        duplicateUser.setPassword(DEFAULT_PASSWORD);
+        duplicateUser.setAge(DEFAULT_AGE);
+
+        assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(duplicateUser),
+                "Expected register() to throw exception, but it didn't"
+        );
+
+        int userCount = 0;
+        for (User user : Storage.people) {
+            if (user.getLogin().equals(DEFAULT_LOGIN)) {
+                userCount++;
+            }
+        }
+        assertEquals(1, userCount, "Duplicate user should not be registered");
+    }
+
+    @Test
+    void ifLoginLessThanSixCharacters_NotOk() {
+        user.setLogin(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifPasswordLessThanSixCharacters_NotOk() {
+        user.setPassword(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifAgeLessThan18_NotOk() {
+        user.setAge(UNDERAGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullLogin_NotOk() {
+        user.setLogin(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullPassword_NotOk() {
+        user.setPassword(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testNullAge_NotOk() {
+        user.setAge(NULL_INTEGER);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void testAgeZero_NotOk() {
+        user.setAge(ZERO_AGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertFalse(registrationService.isSuccesfullyRegistered(user));
+    }
+
+    @Test
+    void ifLoginExactlySixCharacters_Ok() {
+        user.setLogin(SIX_CHARACTER_STRING);
+        registrationService.register(user);
+        assertTrue(registrationService.isSuccesfullyRegistered(user));
```

Comment:
The test 'ifLoginExactlySixCharacters_Ok' should be renamed to reflect the correct requirement for login length if there is one. If the six-character requirement is not part of the task, adjust the test accordingly.